applications/ShallowWaterApplication/custom_processes/write_from_sw_at_interface_process.h

```diff
@@ -0,0 +1,249 @@
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of WriteFromSwAtInterfaceProcess
+    KRATOS_CLASS_POINTER_DEFINITION(WriteFromSwAtInterfaceProcess);
+
+    /// Definition of the node type
+    using NodeType = Node<3>;
+
+    /// Definition of the geometry type
+    using GeometryType = Geometry<NodeType>;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor
+     * @details Removed
+     */
+    WriteFromSwAtInterfaceProcess() = delete;
+
+    /**
+     * @brief Constructor with Model and Parameters
+     */
+    WriteFromSwAtInterfaceProcess(Model& rModel, Parameters ThisParameters = Parameters());
+
+    /**
+     * @brief Destructor
+     */
+    ~WriteFromSwAtInterfaceProcess() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Execute() override;
+
+    int Check() override;
+
+    const Parameters GetDefaultParameters() const override;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override {
+        std::stringstream buffer;
+        buffer << "WriteFromSwAtInterfaceProcess";
+        return buffer.str();
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override {
+        rOStream << Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override {}
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+
+    ///@}
+
+private:
+    ///@name Member Variables
+    ///@{
+
+    ModelPart& mrVolumeModelPart;
+    ModelPart& mrInterfaceModelPart;
+    array_1d<double,3> mDirection;
+    bool mStoreHistorical;
+
+    // Option to print a debug file
+    bool mPrintVelocityProfile;
+
+    // Option to substitute the boundaries by the neighbor
+    bool mExtrapolateBoundaries;
+    NodeType::Pointer mpFirstBoundaryNode;
+    NodeType::Pointer mpSecondBoundaryNode;
+    NodeType::Pointer mpFirstBoundaryNeighbor;
+    NodeType::Pointer mpSecondBoundaryNeighbor;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    // void GetBoundingVolumeLimits(double& rMin, double& rMax);
+
+    void ReadAndSetValues(
+        NodeType& rNode,
+        BinBasedFastPointLocator<TDim>& rLocator,
+        typename BinBasedFastPointLocator<TDim>::ResultContainerType& rResults);
+
+    // array_1d<double,3> InterpolateVelocity(
+    //     const Element::Pointer ElementId,
+    //     const Vector& rShapeFunctionValues) const;
+
+    template<class TDataType, class TVarType = Variable<TDataType>>
+    void SetValue(NodeType& rNode, const TVarType& rVariable, TDataType rValue)
+    {
+        if (mStoreHistorical)
+            rNode.FastGetSolutionStepValue(rVariable) = rValue;
```

Comment:
if are always vectors better use noalias

Reply:
No, they are not always vectors.